lib/ach_client/providers/sftp/ach_transaction.rb

```diff
@@ -13,20 +13,25 @@ def do_send
       # @return [ACH::EntryDetail] ACH gem's ach transaction record
       def to_entry_detail
         entry = ACH::EntryDetail.new
-        entry.transaction_code = transaction_code
-        entry.routing_number = routing_number
-        entry.account_number = account_number
+        entry.transaction_code = remove_newlines(transaction_code)
+        entry.routing_number = remove_newlines(routing_number)
+        entry.account_number = remove_newlines(account_number)
         entry.amount = amount_in_cents
-        entry.individual_id_number = external_ach_id # Doesn't need to be a number
-        entry.individual_name = merchant_name
-        entry.originating_dfi_identification =
+        entry.individual_id_number = remove_newlines(external_ach_id) # Doesn't need to be a number
+        entry.individual_name = remove_newlines(merchant_name)
+        entry.originating_dfi_identification = remove_newlines(
           self.class.parent.originating_dfi_identification
+        )
         entry.trace_number = external_ach_id.gsub(/\D/, '').to_i # Does need to be a number
         entry
       end
 
       private
 
+      def remove_newlines(value)
+        value.gsub(/[\n\r]/, '')
```

Comment:
@zcotter This method pretty much self-documenting, but would you mind adding a comment explaining the intended behavior of the regex?

Reply:
Heh, always comment regexes, I agree.